bq-connector/docai_bq_connector/doc_ai_processing/Processor.py

```diff
@@ -192,16 +192,27 @@ def _process_async(self) -> Union[DocumentOperation, ProcessedDocument]:
         )
 
         operation = client.batch_process_documents(request)
+        logging.debug(f"DocAI Batch Process started. LRO = {operation.operation.name}")
 
         if self.should_async_wait is False:
             return DocumentOperation(operation.operation.name)
 
         # Wait for the operation to finish
         operation.result(timeout=self.async_timeout)
+        logging.debug("DocAI Batch Process finished")
+
+        if operation.metadata and operation.metadata.individual_process_statuses:
+            cur_process_status = operation.metadata.individual_process_statuses[0]
```

Comment:
I actually think it's possible for there to be multiple processes if multiple documents were used in batch processing. Have you tested this to ensure that all processes are handled when multiple documents are processed?

Reply:
Saw your TODO comment